crypto/stark-verifier-ethereum/contracts/merkle.sol

```diff
@@ -0,0 +1,122 @@
+pragma solidity ^0.6.4;
+
+import './hashing.sol';
+import './ring_buffer.sol';
+
+contract MerkleVerifier is Hashing, RingBuffer {
+
+    function verify_merkle_proof(
+        bytes32 root,
+        bytes32[] memory data_points,
+        uint[] memory indices,
+        bytes32[] memory decommitment)
+    internal pure returns(bool) {
+        IndexRingBuffer memory buffer = IndexRingBuffer({
+            front: 0,
+            back: data_points.length-1,
+            indexes: indices,
+            data: data_points,
+            is_empty: false
+        });
+        uint decommitment_index = 0;
+
+        while (true) {
+           (uint index, bytes32 current_hash) = remove_front(buffer);
+
+           uint parent = index/2;
+
+           if (parent > 0) {
```

Comment:
Looks cleaner (less indented code) to turn this into an exit condition:

```
if (index == 1) {
    return(root == current_hash);
}
```

Doing it directly on `index` avoids creating an invalid `parent` index.